--- src/tModLoader/Terraria.ModLoader/ModLoader.cs
+++ src/64bit/Terraria.ModLoader/ModLoader.cs
@@ -18,6 +_,7 @@
 using Terraria.ModLoader.Default;
 using Terraria.ModLoader.Engine;
 using Terraria.ModLoader.UI;
+using Terraria.ModLoader.x64bit;
 using Version = System.Version;
 
 namespace Terraria.ModLoader
@@ -51,7 +_,8 @@
 		[Obsolete("Use Platform.IsLinux")]
 		public static readonly bool linux = Platform.IsLinux;
 		[Obsolete("Use Platform.IsOSX")]
+		//Better having a field to check if the process is in 64bit mode
-		public static readonly bool mac = Platform.IsOSX;
+		public static readonly bool is64BitProcess = Environment.Is64BitProcess;
 
 		[Obsolete("Use CompressedPlatformRepresentation instead")]
 		public static readonly string compressedPlatformRepresentation = Platform.IsWindows ? "w" : (Platform.IsLinux ? "l" : "m");
@@ -360,6 +_,9 @@
 			Main.Configuration.Put("AvoidImgur", UI.ModBrowser.UIModBrowser.AvoidImgur);
 			Main.Configuration.Put(nameof(UI.ModBrowser.UIModBrowser.EarlyAutoUpdate), UI.ModBrowser.UIModBrowser.EarlyAutoUpdate);
 			Main.Configuration.Put("LastLaunchedTModLoaderVersion", version.ToString());
+			
+			// tModLoader 64 bit specific Configs
+			Main.Configuration.Put("tMod64_XlWorldsDisabled", Core64.xlWorldsDisabled);
 		}
 
 		internal static void LoadConfiguration()
@@ -377,6 +_,15 @@
 			Main.Configuration.Get("AvoidImgur", ref UI.ModBrowser.UIModBrowser.AvoidImgur);
 			Main.Configuration.Get(nameof(UI.ModBrowser.UIModBrowser.EarlyAutoUpdate), ref UI.ModBrowser.UIModBrowser.EarlyAutoUpdate);
 			LastLaunchedTModLoaderVersion = new Version(Main.Configuration.Get("LastLaunchedTModLoaderVersion", "0.0"));
+			
+			// tModLoader 64 bit specific Configs
+			Main.Configuration.Get("tMod64_XlWorldsDisabled", ref Core64.xlWorldsDisabled);
+#if CLIENT
+			if (Core64.xlWorldsDisabled) {
+				Main.tile = new Tile[8401, 2401];
+				Logging.Terraria.Info("Resized Main.tile array to 8401x2401");
+			}
+#endif
 		}
 
 		internal static void MigrateSettings()
